tethysext/atcore/models/resource_workflow_steps/spatial_input_rws.py

```diff
@@ -123,7 +123,7 @@ def validate_feature_attributes(self, attributes):
             # Skip attributes that are not given, raising a validation error if it is required
             if attribute_name not in attributes:
                 if value_required:
-                    validation_errors.append(f'{attribute_title} is required.')
+                    validation_errors.append(f'{defined_attribute.label} is required.')
```

Comment:
Instead of changing `attribute_title` in multiple places, change the definition of `attribute_title` on line 121. Also, `label` may not be defined, so fallback to the `attribute_name.replace(...).title()` if it isn't.